### PR TITLE
Move invite parent user option to separate wizard from add user wizard

### PR DESCRIPTION
**Pull request by PasinduYeshan**
### Purpose
$subject

https://github.com/wso2/identity-apps/assets/61885844/d04cfc2e-04d6-41b9-8347-5265a31d380a

### Related Issues
- https://github.com/wso2/product-is/issues/18633
- https://github.com/wso2/product-is/issues/18553

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    IdentifiableComponentInterface,
+    TestableComponentInterface
+} from "@wso2is/core/models";
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:22]: Let's keep this in a single line. Also, we don't need both. just IdentifiableComponentInterface would be enough as TestableComponentInterface is deprecated.

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Grid, Modal } from "semantic-ui-react";
+import { UserManagementConstants } from "../../constants";
+import { InviteParentOrgUser } from "../guests/pages/invite-parent-org-user";
+
+/**
+ * Interface for the Invite Parent Org User Wizard.
+ */
+interface InviteParentOrgUserWizardProps extends IdentifiableComponentInterface, TestableComponentInterface {
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:33]:
```suggestion
interface InviteParentOrgUserWizardProps extends IdentifiableComponentInterface {
```

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+export const InviteParentOrgUserWizard: FunctionComponent<InviteParentOrgUserWizardProps> = (
+    props: InviteParentOrgUserWizardProps
+): ReactElement => {
+
+    const {
+        closeWizard,
+        onUserInviteSuccess,
+        [ "data-testid" ]: testId
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:48]: Let's use componentid as using testid has been deprecated.

```suggestion
        [ "data-componentid" ]: componentId
```

File: apps/console/src/public/deployment.config.json [hunk 4]
```diff
@@ -604,7 +604,6 @@
                         "internal_guest_mgt_invite_list"
                     ],
                     "create": [
-                        "internal_guest_mgt_invite_validate",
```
Comment (pavinduLakshan) [inline apps/console/src/public/deployment.config.json:607]: Let's update the framework, j2 template, local configs for cloud development, cloud pipeline and the newly introduced api-resources.xml as well 😪

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    const { t } = useTranslation();
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+    const [ alert, setAlert, alertComponent ] = useWizardAlert();
+
+    /**
+     * Triggers a form submit event for the form in the InviteParentOrgUser component.
+     */
+    const submitParentUserInviteForm = () => {
+        document
+            .getElementById(UserManagementConstants.INVITE_PARENT_ORG_USER_FORM_ID)
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:60]: Directly accessing browser primitives in React components is an anti-pattern. We should be able to handle form submission in a more "React" way. Could u please check the similar implementations in other places and modify this accordingly?

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            .dispatchEvent(new Event("submit", { bubbles: true, cancelable: true }));
+    };
+
+    return (
+        <Modal
+            data-testid={ testId }
+            open={ true }
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:67]:
```suggestion
            open
```

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                    closeWizard={ closeWizard }
+                    setIsSubmitting={ setIsSubmitting }
+                    onUserInviteSuccess={ onUserInviteSuccess }
+                    setAlert={ setAlert }
+                />
+            </Modal.Content>
+            <Modal.Actions>
+                <Grid>
+                    <Grid.Row column={ 1 }>
+                        <Grid.Column mobile={ 8 } tablet={ 8 } computer={ 8 }>
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:93]:
```suggestion
                        <Grid.Column mobile={ 8 }>
```

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+                                floated="left"
+                                onClick={ () => {
+                                    closeWizard();
+                                } }
+                            >
+                                { t("common:cancel") }
+                            </LinkButton>
+                        </Grid.Column>
+                        <Grid.Column mobile={ 8 } tablet={ 8 } computer={ 8 }>
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:105]:
```suggestion
```

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                            <PrimaryButton
+                                data-testid={ `${ testId }-finish-button` }
+                                floated="right"
+                                onClick={ submitParentUserInviteForm }
+                                loading={ isSubmitting }
+                                disabled={ isSubmitting }
+                            >
+                                { t("extensions:manage.features.user.addUser.invite") }
+                            </PrimaryButton>
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:115]:
```suggestion
```

File: apps/console/src/features/users/pages/users.tsx [hunk 10]
```diff
@@ -753,32 +756,65 @@ const UsersPage: FunctionComponent<UsersPageInterface> = (
...
-            text: t("console:manage.features.users.addUserDropDown.addNewUser"),
-            value: UserAccountTypesMain.EXTERNAL
-        },
-        {
-            "data-componentid": `${ testId }-bulk-import-users-dropdown-item`,
-            "data-testid": `${ testId }-bulk-import-users-dropdown-item`,
-            key: 2,
-            text: t("console:manage.features.users.addUserDropDown.bulkImport"),
-            value: UserAddOptionTypes.BULK_IMPORT
+    const addUserOptions = (): DropdownItemProps[] => {
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/pages/users.tsx:759]:
```suggestion
    const getAddUserOptions = (): DropdownItemProps[] => {
```

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 2 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx]: Changed

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 3 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx]: Changed

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 6 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx]: Changed

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 8 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx]: Removed

File: apps/console/src/features/users/pages/users.tsx (hunk 10 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/pages/users.tsx]: Changed

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 9 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx]: Removed

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 7 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx]: Removed

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 5 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx:58]: https://codesandbox.io/s/1y7noyrlmq
According to the `FinalForm` documentation, this is how external submit is handled.

File: apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx (hunk 1 above)
Comment (PasinduYeshan) [reply apps/console/src/features/users/components/wizard/invite-parent-org-user-wizard.tsx]: Fixed

File: apps/console/src/public/deployment.config.json (hunk 4 above)
Comment (PasinduYeshan) [reply apps/console/src/public/deployment.config.json:607]: Framework updates - https://github.com/wso2/carbon-identity-framework/pull/5314
This feature is not in cloud deployment yet. Will mention the config update requirement in the relevant issue.

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**